b3sum/src/main.rs

```diff
@@ -166,31 +185,46 @@ fn main() -> Result<()> {
         blake3::Hasher::new()
     };
     let print_names = !args.is_present(NO_NAMES_ARG);
+    let raw_output = args.is_present(RAW_ARG);
     let mut did_error = false;
+
     if let Some(files) = args.values_of_os(FILE_ARG) {
-        for filepath in files {
-            let filepath_str = filepath.to_string_lossy();
-            match hash_file(&base_hasher, filepath) {
-                Ok(output) => {
-                    write_output(output, len)?;
-                    if print_names {
-                        println!("  {}", filepath_str);
-                    } else {
-                        println!();
+        if raw_output && files.len() > 1 {
```

Comment:
Rather than indenting the main loop by another level, let's `bail!` if this fails.

Reply:
done